pandas/compat/_optional.py

```diff
@@ -92,10 +93,16 @@ def import_optional_dependency(
             raise ImportError(msg) from None
         else:
             return None
-
+    # Grab parent module if submodule being imported
```

Comment:
why are you changing this?

Reply:
Pyarrow's csv module has to be imported as a submodule and import_optional_dependency doesn't handle submodules correctly.

Reply:
can you add a comment to this effect